instrumentation/active_support/test/opentelemetry/instrumentation/active_support/span_subscriber_test.rb

```diff
@@ -205,123 +200,238 @@ def finish(name, id, payload)
+            ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+
+            _(last_span).wont_be_nil
+            _(last_span.name).must_equal(notification_name)
+            _(last_span.attributes['extra']).must_equal('context')
+          end
+        end
+
+        describe 'when using a unstable formatter' do
+          it 'defaults to the notification name' do
+            allow(OpenTelemetry).to receive(:handle_error).with(exception: RuntimeError, message: String)
+
+            OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_pattern, nil, nil, span_name_formatter: ->(_) { raise 'boom' })
+            ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+
+            _(last_span).wont_be_nil
+            _(last_span.name).must_equal(notification_name)
+            _(last_span.attributes['extra']).must_equal('context')
+          end
+        end
+      end
+
+      it 'finishes spans even when block subscribers blow up' do
+        # This scenario cannot be exercised reliably on Active Support < 7.0 since the #finish method
+        # will never be called by the notifier if another subscriber raises an error.
+        #
+        # See this PR for additional details: https://github.com/rails/rails/pull/43282
+        skip 'Notifications will be broken in this scenario on Active Support < 7.0' if ActiveSupport.version < Gem::Version.new("7.0")
+
+        ActiveSupport::Notifications.subscribe(notification_pattern) { raise 'boom' }
+        OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_pattern)
+
+        expect do
+          ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+        end.must_raise RuntimeError
+
+        _(last_span).wont_be_nil
+        _(last_span.name).must_equal(notification_name)
+        _(last_span.attributes['extra']).must_equal('context')
+      end
+
+      it 'finishes spans even when complex subscribers blow up' do
+        # This scenario cannot be exercised reliably on Active Support < 7.0 since the #finish method
+        # will never be called by the notifier if another subscriber raises an error.
+        #
+        # See this PR for additional details: https://github.com/rails/rails/pull/43282
+        skip 'Notifications will be broken in this scenario on Active Support < 7.0' if ActiveSupport.version < Gem::Version.new("7.0")
+
+        ActiveSupport::Notifications.subscribe(notification_pattern, CrashingEndSubscriber.new)
+        OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_pattern)
+
+        expect do
+          ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+        end.must_raise RuntimeError
+
+        _(last_span).wont_be_nil
+        _(last_span.name).must_equal(notification_name)
+        _(last_span.attributes['extra']).must_equal('context')
+      end
```

Comment:
I had to skip these two tests on Active Support 6.1 since there's a bug in this version that prevents the subscriber from properly finishing the spans if another subscriber raises an error.

See https://github.com/rails/rails/pull/43282 for further details.